client/src/pages/ContestDetails.js

```diff
@@ -223,13 +269,7 @@ export default function ContestDetails(props) {
                                 </div>
                             </Grid>
                             <Grid item xs={5} className={classes.submitButtonDiv}>
-                                <Link to={{
-                                        pathname: `/submit-design/${contestId}`
-                                }}>
-                                    <Button variant='outlined' className={classes.submitDesign}>
-                                        Submit Design
-                                    </Button>
-                                </Link>
+                                {submitButton}
```

Comment:
Instead of creating a state to render this (states should ideally not be components), you can just gate this off, introducing a `isOwner` state instead

```
{!isOwner && <SubmitButton />
```
as an example